pkg/operators/wasm/wasm.go

```diff
@@ -274,11 +276,47 @@ func (i *wasmOperatorInstance) init(
 		return fmt.Errorf("unsupported gadget API version: %d, expected: %d", ret[0], apiVersion)
 	}
 
+	// add extra info to gadgetcontext if requested
+	if gadgetCtx.ExtraInfo() {
+		err := i.addExtraInfo(gadgetCtx, ret[0], wasmProgram)
+		if err != nil {
+			return fmt.Errorf("adding extra info: %w", err)
+		}
+	}
+
 	i.dataSourceCallback = mod.ExportedFunction("dataSourceCallback")
 
 	return err
 }
 
+func (i *wasmOperatorInstance) addExtraInfo(gadgetcontext operators.GadgetContext, version uint64, wasmProgram []byte) error {
+	module, err := i.rt.CompileModule(gadgetcontext.Context(), wasmProgram)
+	if err != nil {
+		return err
+	}
+	imports := module.ImportedFunctions()
+	upcalls := []string{}
+	for _, imp := range imports {
+		parts := strings.Split(imp.Name(), ".")
+		upcalls = append(upcalls, parts[len(parts)-1])
```

Comment:
This gets both upcalls ("setConfig","kallsymsSymbolExists","gadgetLog") and other functions that are not provided by IG such as `runtime.fd_write` or `__imported_wasi_snaps`.

You can filter those functions with:

```go
moduleName, name, isImport := imp.Import()
```
If `isImport` is true and `moduleName` is "ig", then it should be an upcall. But it needs to be rebased on #4209.